### DISCUSSION TITLE
Get rid accept cookies popup for StackOverflow when using CLI

### DISCUSSION THREAD

**Discussion by Ken0sis** [gildas-lormeau/SingleFile#747] [Q&A]
**Is your feature request related to a problem? Please describe.**
When using the CLI, html capture will show cookies popup, and this blocks part of the Stackoverflow page.

**Describe the solution you'd like**
Can you please share a script that will close the cookies popup for Stackoverflow?

You've helped me with a similar problem with Reddit threads not unfolding, and I thought I could easily write a script to close the cookies popup, but I'm not having luck.  If there's a general script that can close all cookies popup, I think that would add a lot of value to the CLI for everyone, because that would make the output look more like what we get when using web browser extension.  Thank you for considering your help.

**Comment by WellRedPandit** [comment]
> If there's a general script that can close all cookies popup, I think that would add a lot of value to the CLI for everyone

so true! doubt it's possible, though... some sites demand using mouse and ignore keys, some use a delay and you have to know what to wait for... it's almost like a plugin and its own project are needed to keep track of sites and their popups... i don't know if singlefile has a plugin architecture in place currently to develop something like that independently from core singlefile itself...

**Comment by Ken0sis** [comment]
I'm wondering if anyone has any luck writing a script to get rid of the popup for just Stackoverflow?  It's such a heavily viewed and referenced site.

**Comment by gildas-lormeau** [comment]
Sorry for the late answer, I guess running this [userscript](https://github.com/gildas-lormeau/SingleFile/wiki/How-to-execute-a-user-script-before-a-page-is-saved) should do the job:
```js
onload = () => {
  document.querySelector(".js-accept-cookies").click();
};
```

**Comment by ozanguven** [comment]
I tried to close popups and accept cookie banners with SingleFile CLI, but failed.

How should we use user scripts? I tried with below command and [sample codes](https://github.com/gildas-lormeau/SingleFile/wiki/How-to-execute-a-user-script-before-a-page-is-saved), but it does not work.

`docker run singlefile --browser-script=/path/to/scripts.js 'https://www.google.com/'`

**Comment by gildas-lormeau** [comment]
The script I provided is specific to StackOverflow.

For google.com, you could run this script instead:
```js
onload = () => {
  document.evaluate("//*[. = 'I agree']", document.body).iterateNext().click();
};
```

**Comment by ozanguven** [comment]
Thanks a lot for your support and maintaining SingleFile. I was using Docker, and could not managed to run user scripts. I made a manual installation as described [here](https://github.com/gildas-lormeau/SingleFile/tree/master/cli) and it works great.

I can provide another example to close OneTrust cookie banner:

```
(() => {

addEventListener("single-file-on-before-capture-request", () => {
   document.evaluate('//*[@id="onetrust-accept-btn-handler"]', document.body).iterateNext().click();
});

})();
```
My command:
`single-file --browser-executable-path=/usr/bin/google-chrome-stable --browser-script=/path/to/scripts.js 'https://website-url' file-name.html`

**Comment by gildas-lormeau** [comment]
I'm moving this issue in the discussion tab because I think and hope it's a better place for it.
